build_tools/github_actions/post_build_upload.py

```diff
@@ -0,0 +1,301 @@
+#!/usr/bin/env python3
+
+"""
+This script runs after building TheRock, where this script does:
+1. Create log archives
+2. Create log index files
+3. (optional) upload artifacts
+4. (optional) upload logs
+5. (optional) add links to GitHub job summary
+"""
+
+import argparse
+import os
+import tarfile
+from pathlib import Path
+import platform
+import shlex
+import shutil
+import subprocess
+import sys
+
+THEROCK_DIR = Path(__file__).resolve().parent.parent.parent
+PLATFORM = platform.system().lower()
+
+# Importing indexer.py
+sys.path.append(str(THEROCK_DIR / "third-party" / "indexer"))
+from indexer import process_dir
+from github_actions_utils import *
+
+
+def log(*args):
+    print(*args)
+    sys.stdout.flush()
+
+
+def exec(cmd: list[str], cwd: Path):
+    log(f"++ Exec [{cwd}]$ {shlex.join(cmd)}")
+    subprocess.run(cmd, check=True)
+
+
+def is_windows():
+    return platform.system().lower() == "windows"
+
+
+def normalize_path(p: Path) -> str:
+    return str(p).replace("\\", "/") if is_windows() else str(p)
```

Comment:
This code has moved around a few times. I still think we can replace it with just `p.as_posix()` at the call site, instead of having this string processing code in our own helper function.